ministryofjustice/correspondence_tool_staff
# [04] CT 2192 Add report download operations (PR No. 4)

## Description
Closed Cases are generated using the new ETL pipeline which generates the report on a separate process/container, zipped and then made available for download with transfer from the separate process/container to the web-app via Redis. This makes closed cases download asynchronous, and results in a file download of ~1 MB rather than (at present) 15mb.
 
## Self-review checklist

 

### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2192
 
### Deployment
Requires `rails db:migrate`, ensure Report Type 007 has `etl = true`
  
### Manual testing instructions
1. Go to 'closed cases' in the web app and click 'download cases'. Enter dates as usual
2. Press submit and a zip file should be generated for download
